### ISSUE TITLE
converter.parse throws KeyError parsing midifile

### ISSUE THREAD

**Issue by smirkingman** [cuthbertLab/music21#1530]
**music21 version**

`8.1.0`   

**Problem summary**

When parsing the atttached MIDI file KeyError gets thrown:

>   File "D:\projects\music\MTSpectralMozartConcertos\midiscore.py", line 22, in __init__
    self.score = converter.parse(self.midi_path, quarterLengthDivisors=(8, 3))
  File "D:\projects\Music\venv\lib\site-packages\music21\converter\__init__.py", line 1297, in parse
    return parseFile(valueStr, number=number, format=format,
  File "D:\projects\Music\venv\lib\site-packages\music21\converter\__init__.py", line 1151, in parseFile
    v.parseFile(fp, number=number, format=format, forceSource=forceSource, **keywords)
  File "D:\projects\Music\venv\lib\site-packages\music21\converter\__init__.py", line 621, in parseFile
    self.parseFileNoPickle(fp, number, format, forceSource, **keywords)
  File "D:\projects\Music\venv\lib\site-packages\music21\converter\__init__.py", line 543, in parseFileNoPickle
    self.subConverter.parseFile(
  File "D:\projects\Music\venv\lib\site-packages\music21\converter\subConverters.py", line 1203, in parseFile
    midiTranslate.midiFilePathToStream(filePath, inputM21=self.stream, **keywords)
  File "D:\projects\Music\venv\lib\site-packages\music21\midi\translate.py", line 2705, in midiFilePathToStream
    return midiFileToStream(mf, inputM21=inputM21, **keywords)
  File "D:\projects\Music\venv\lib\site-packages\music21\midi\translate.py", line 2876, in midiFileToStream
    midiTracksToStreams(mf.tracks,
  File "D:\projects\Music\venv\lib\site-packages\music21\midi\translate.py", line 2613, in midiTracksToStreams
    midiTrackToStream(mt,
  File "D:\projects\Music\venv\lib\site-packages\music21\midi\translate.py", line 2091, in midiTrackToStream
    s.makeTies(inPlace=True)
  File "D:\projects\Music\venv\lib\site-packages\music21\stream\base.py", line 6499, in makeTies
    return makeNotation.makeTies(
  File "D:\projects\Music\venv\lib\site-packages\music21\stream\makeNotation.py", line 1312, in makeTies
    dst.insert(0, eRemain)
  File "D:\projects\Music\venv\lib\site-packages\music21\stream\base.py", line 2257, in insert
    storeSorted = self.coreInsert(offset,
  File "D:\projects\Music\venv\lib\site-packages\music21\stream\core.py", line 111, in coreInsert
    thisSortTuple = element.sortTuple().modify(offset=offset)
  File "D:\projects\Music\venv\lib\site-packages\music21\base.py", line 2672, in sortTuple
    insertIndex = self.sites.siteDict[id(self.activeSite)].globalSiteIndex

but there is no 'raise' there at line 2672:
![ss](https://user-images.githubusercontent.com/22235329/220633119-0dfed113-53ce-4768-95b1-cd11eb0b4d6e.jpg)

**Steps to reproduce**

```
`score = converter.parse([attached.mid], quarterLengthDivisors=(8, 3))`
```
**Expected vs. actual behavior**

This MIDI file reads sucessfully with MuseScore, VLC, etc.
I have several MIDI files that fail like this.

[K414 Piano Concerto No.12 in A_K.414 - 2.Andante (after J.C. Bach).zip](https://github.com/cuthbertLab/music21/files/10804135/K414.Piano.Concerto.No.12.in.A_K.414.-.2.Andante.after.J.C.Bach.zip)

**More information**

Windows 10, PyCharm, Python 3.9.0

**Comment by mscuthbert** [comment]
Hi -- does it parse properly if quarterLengthDivisors is not set?  what if `forceSource=True` is given.

Unfortunately, I'm not going to be able to diagnose an entire piano concerto -- can you use a debugger to figure out which track is causing the problem?  and from there which measure(s)?

We recommend converting MIDI in another software to MusicXML before parsing in music21 -- our quantizer is not up to the standard of Finale or MuseScore.

**Comment by smirkingman** [comment]
Thanks for the prompt follow-up!

Neither removing qLD nor adding forceSource fixes it.

I'd tried to debug, but the strange thing was that KeyError was thrown  at line 2672, in sortTuple, which isn't a 'raise' statement, so I was stumped.

I have quite a number of midi files like this and I only have MuseScore; looks like I'll have to bite the bullet and convert everything to MusicXML. Damn!
>;-)

**Comment by mscuthbert** [comment]
With the mscore executable within MuseScore directory, you can script the conversion to midi.

**Comment by jacobtylerwalls** [comment]
Interestingly, this file no longer fails on `main`. I bisected the fix (easy peasy with `git bisect run`) to 86d4cf652338c0f4db6ac2c561c29a433c916e2f. So the fix should be in with the (soon to be released) v9.

**Comment by gg22mm** [comment]
The same problem, for multiple files, an error will be reported / 一样的问题，for 多文件就会报错
